test_metadata.py

```diff
@@ -0,0 +1,78 @@
+import unittest
+import uuid
+import metadata.picture_set as picture_set_data
+import metadata.picture as picture_data
+import metadata.validator as validator
+import io
+from PIL import Image
+from datetime import date
+import base64
+import json
+
+PIC_LINK = 'test.com'
+
+PIC_PATH = 'test_image.tiff'
+
+class test_picture_functions(unittest.TestCase):
+    def setUp(self):
+        self.image = Image.new('RGB', (1980, 1080),'blue')
+        self.image_byte_array = io.BytesIO()
+        self.image.save(self.image_byte_array, format='TIFF')
+        self.pic_encoded = base64.b64encode(self.image_byte_array.getvalue()).decode("utf8")
+        self.link = PIC_LINK
+        self.nb_seeds = 1
+        self.zoom = 1.0
+    
+    def test_build_picture(self):
+        """
+        This test checks if the build_picture function returns a valid JSON object
+        """
+        picture = picture_data.build_picture(self.pic_encoded, self.link, self.nb_seeds, self.zoom)
+        properties = picture_data.get_image_properties(self.pic_encoded)
+        data = json.loads(picture)
+        
+        self.assertEqual(data["user_data"]["description"], "", "Description should be empty")
+        self.assertEqual(data["user_data"]["number_of_seeds"], 1, "Number of seeds should be 1")
+        self.assertEqual(data["user_data"]["zoom"], 1.0, "Zoom should be 1.0")
+        self.assertEqual(data["metadata"]["upload_date"], str(date.today()), "Upload date should be today's date")
+        self.assertEqual(data["image_data"]["format"], properties[2], "Format should be 'TIFF'")
+        self.assertEqual(data["image_data"]["height"], properties[1], "Height should be 1080 as an integer")
+        self.assertEqual(data["image_data"]["width"], properties[0],"Width should be 1980 as an integer")
+        self.assertEqual(data["image_data"]["resolution"], "", "Resolution should be empty")
+        self.assertEqual(data["image_data"]["source"], PIC_LINK,f"Source should be {PIC_LINK}")
+        self.assertEqual(data["quality_check"]["image_checksum"], "", "Checksum should be empty")
+        self.assertEqual(data["quality_check"]["upload_check"], True, "Upload check should be True")
+        self.assertEqual(data["quality_check"]["valid_data"], True, "Valid data should be True")
+        self.assertEqual(data["quality_check"]["error_type"], "", "Error type should be empty")
+        self.assertEqual(data["quality_check"]["quality_score"], 0.0, "Quality score should be 0.0")
+        
```

Comment:
Instead of having the long list of check, you can have a mock or expected data dictionary with all the expected value and then compare data to the expected data